File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen={}
        #key - num
        #value - indice
        for i in range(len(nums)):
            if (target - nums[i]) in seen.keys():
                return [seen.get(target - nums[i]), i]
            seen[nums[i]] = i
        return None

        

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size();
        vector<int> result(nums.size(), 1);

        for(int i = 0; i<n; ++i){
            for(auto j = 0; j<n; ++j){
                if (i!=j){
                    result[i]*=nums[j];
                }
            }
        }
        return result;

    }
};
